xbmc/music/infoscanner/MusicInfoScanner.cpp

```diff
@@ -112,7 +104,14 @@ void CMusicInfoScanner::Process()
         m_handle = dialog->GetHandle(g_localizeStrings.Get(314));
     }
 
-    m_bClean = g_advancedSettings.m_bMusicLibraryCleanOnUpdate;
+    // check if we only need to perform a cleaning
+    if (m_bClean && m_pathsToScan.empty())
+    {
+      CleanDatabase(false);
+      m_bRunning = false;
+
+      return;
```

Comment:
side effect - m_handle is set. It's inconsequential, I was just pointing out an example of why early returns should be avoided. And it looks like it never actually opens the progress display before returning?